ext/node/polyfill.rs

```diff
@@ -1,229 +1,217 @@
 // Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
 
-use deno_core::error::generic_error;
-use deno_core::error::AnyError;
-use deno_core::url::Url;
-use deno_core::ModuleSpecifier;
-
-// TODO(bartlomieju): seems super wasteful to parse the specifier each time
-pub fn resolve_builtin_node_module(module_name: &str) -> Result<Url, AnyError> {
-  if let Some(module) = find_builtin_node_module(module_name) {
-    return Ok(ModuleSpecifier::parse(module.specifier).unwrap());
-  }
-
-  Err(generic_error(format!(
-    "Unknown built-in \"node:\" module: {module_name}"
-  )))
-}
-
-fn find_builtin_node_module(module_name: &str) -> Option<&NodeModulePolyfill> {
+/// e.g. `is_builtin_node_module("assert")`
+pub fn is_builtin_node_module(module_name: &str) -> bool {
   SUPPORTED_BUILTIN_NODE_MODULES
     .iter()
-    .find(|m| m.name == module_name)
-}
-
-pub fn is_builtin_node_module(module_name: &str) -> bool {
-  find_builtin_node_module(module_name).is_some()
+    .any(|m| m.module_name() == module_name)
 }
 
 pub struct NodeModulePolyfill {
   /// Name of the module like "assert" or "timers/promises"
-  pub name: &'static str,
   pub specifier: &'static str,
+  pub ext_specifier: &'static str,
+}
+
+impl NodeModulePolyfill {
+  pub fn module_name(&self) -> &'static str {
+    debug_assert!(self.specifier.starts_with("node:"));
+    &self.specifier[5..]
```

Comment:
strip_prefix and unwrap

Reply:
It was previously that https://github.com/denoland/deno/pull/19040#discussion_r1203158425

I think either way is fine because all of these should start with "node:"